PhysicsTools/NanoAOD/plugins/TriggerObjectTableProducer.cc

```diff
@@ -91,8 +91,38 @@ class TriggerObjectTableProducer : public edm::stream::EDProducer<> {
           l1DR2_2(-1),
           l2DR2(-1),
           skipObjectsNotPassingQualityBits(pset.getParameter<bool>("skipObjectsNotPassingQualityBits")),
-          qualityBits(pset.getParameter<std::string>("qualityBits")),
-          qualityBitsDoc(pset.getParameter<std::string>("qualityBitsDoc")) {
+          qualityBits("0"),   //will be overwritten from configuration
+          qualityBitsDoc("")  //will be created from configuration
+    {
+      if (pset.existsAs<std::string>("qualityBits")) {
+        qualityBits = StringObjectFunction<pat::TriggerObjectStandAlone>(pset.getParameter<std::string>("qualityBits"));
+        qualityBitsDoc = pset.getParameter<std::string>("qualityBitsDoc");
+      } else {
+        std::vector<edm::ParameterSet> qualityBitsConfig =
+            pset.getParameter<std::vector<edm::ParameterSet>>("qualityBits");
+        std::stringstream qualityBitsFunc;
+        std::vector<bool> bits(qualityBitsConfig.size(), false);
+        for (size_t i = 0; i != qualityBitsConfig.size(); ++i) {
+          if (i != 0) {
+            qualityBitsFunc << " + ";
+            qualityBitsDoc += ", ";
+          }
+          unsigned int bit = i;
+          if (qualityBitsConfig[i].existsAs<unsigned int>("bit"))
+            bit = qualityBitsConfig[i].getParameter<unsigned int>("bit");
+          assert(!bits[bit] && "a quality bit was inserted twice");  // the bit should not have been set already
+          assert(bit <= 32 && "quality bits are store on 32 bit");
```

Comment:
shouldn't the check be `bit < 31`? with a signed integer the largest we can represent is `int("1"*31, base=2)`. Also, we could check with `qualityBitsConfig.size()` directly